src/main/java/no/digipost/api/client/internal/delivery/ArchiveDeliverer.java

```diff
@@ -0,0 +1,119 @@
+import static org.apache.commons.lang3.StringUtils.defaultIfBlank;
+
+public class ArchiveDeliverer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ArchiveDeliverer.class);
+
+    private final ArchiveApi apiService;
+    private final EventLogger eventLogger;
+
+    public ArchiveDeliverer(DigipostClientConfig config, ArchiveApi apiService) {
+        this.eventLogger = config.eventLogger.withDebugLogTo(LOG);
+        this.apiService  = apiService;
+    }
+
+    public ArchiveApi.ArchivingDocuments createArchive(Archive archive) {
+        return new PerformingArchivingDocuments(archive, this);
+    }
+
+    /**
+     * Sender melding med alle dokumenter og innhold med én API-forespørsel (HTTP multipart request).
+     */
+    public Archive sendMultipartMessage(Archive archive, Map<UUID, DocumentContent> documentsAndContent) {
+
+        final Map<ArchiveDocument, InputStream> documentInputStream = archive.getDocuments().stream()
+                .collect(toMap(identity(), (doc) -> documentsAndContent.get(doc.getUuid()).getDigipostContent()));
+
+        try {
+            ByteArrayOutputStream bao = new ByteArrayOutputStream();
+            marshal(jaxbContext, archive, bao);
+            ByteArrayBody attachment = new ByteArrayBody(bao.toByteArray(),
+                    ContentType.create(MediaTypes.DIGIPOST_MEDIA_TYPE_V7, UTF_8), "archive");
+
+            MultipartEntityBuilder multipartEntity = MultipartEntityBuilder.create()
+                    .setMode(HttpMultipartMode.STRICT)
+                    .setMimeSubtype(DIGIPOST_MULTI_MEDIA_SUB_TYPE_V7)
+                    .addPart(FormBodyPartBuilder.create("archive", attachment)
+                            .addField("Content-Disposition", "attachment;" + " filename=\"archive\"")
+                            .build());
+
+            for (Entry<ArchiveDocument, InputStream> documentAndContent : documentInputStream.entrySet()) {
+                ArchiveDocument document = documentAndContent.getKey();
+                InputStream content = documentAndContent.getValue();
+
+
+                byte[] bytes = IOUtils.toByteArray(content);
+                multipartEntity = multipartEntity
+                        .addPart(FormBodyPartBuilder
+                                .create("application", new ByteArrayBody(bytes, ContentType.create(defaultIfBlank(document.getContentType(), "application/octet-stream")), document.uuid.toString()))
+                                .addField("Content-Disposition", "attachment;" + " filename=\"" + document.uuid.toString() + "\"").build());
+            }
+            eventLogger.log("*** STARTER INTERAKSJON MED API: Arkiverer filer ***");
+            try (CloseableHttpResponse response = apiService.sendMultipartArchive(multipartEntity.build())) {
+                checkResponse(response, eventLogger);
+
+                eventLogger.log("Brevet ble sendt. Status: [" + response + "]");
```

Comment:
Kan vi kanskje bruke "ArkivDokumentene" eller "Dokumentene" i stedet for "Brevet" her? Ser for meg at det potensielt kan skape litt forvirring :)

Reply:
Jeg forcepushet endringen.